grpc/client/src/main/java/io/helidon/grpc/client/ClientMethodDescriptor.java

```diff
@@ -0,0 +1,562 @@
+         */
+        Config<ReqT, ResT> gauged();
+
+        /**
+         * Collect metrics for this method using {@link org.eclipse.microprofile.metrics.Meter}.
+         *
+         * @return this {@link Config} instance for fluent call chaining
+         */
+        Config<ReqT, ResT> metered();
+
+        /**
+         * Collect metrics for this method using {@link org.eclipse.microprofile.metrics.Histogram}.
+         *
+         * @return this {@link Config} instance for fluent call chaining
+         */
+        Config<ReqT, ResT> histogram();
+
+        /**
+         * Collect metrics for this method using {@link org.eclipse.microprofile.metrics.Timer}.
+         *
+         * @return this {@link Config} instance for fluent call chaining
+         */
+        Config<ReqT, ResT> timed();
+
+        /**
+         * Explicitly disable metrics collection for this service.
+         *
+         * @return this {@link Config} instance for fluent call chaining
+         */
+        Config<ReqT, ResT> disableMetrics();
+
+        /**
+         * Add a {@link Context.Key} and value to be added to the call {@link Context}
+         * when this method is invoked.
+         *
+         * @param key   the {@link Context.Key} to add
+         * @param value the value to map to the {@link Context.Key}
+         * @param <T>   the type of the {@link Context.Key} and value
+         * @return this {@link Config} instance for fluent call chaining
+         * @throws NullPointerException if the key parameter is null
+         */
+        <T> Config<ReqT, ResT> addContextKey(Context.Key<T> key, T value);
+
+        /**
+         * Sets the type of parameter of this method.
+         *
+         * @param type The type of parameter of this method.
+         * @return this {@link Config} instance for fluent call chaining
+         */
+        Config<ReqT, ResT> requestType(Class type);
+
+        /**
+         * Sets the type of parameter of this method.
+         *
+         * @param type The type of parameter of this method.
+         * @return this {@link Config} instance for fluent call chaining
+         */
+        Config<ReqT, ResT> responseType(Class type);
+
+        /**
+         * Register one or more {@link ClientInterceptor interceptors} for the method.
+         *
+         * @param interceptors the interceptor(s) to register
+         * @return this {@link io.helidon.grpc.client.ClientMethodDescriptor.Config} instance for fluent call chaining
+         */
+        Config<ReqT, ResT> intercept(ClientInterceptor... interceptors);
+    }
+
+    /**
+     * {@link MethodDescriptor} builder implementation.
+     *
+     * @param <ReqT> request type
+     * @param <ResT> response type
+     */
+    public static class Builder<ReqT, ResT>
+            implements Config<ReqT, ResT> {
+
+        // This is the currently built ClientMethodDescriptor.
+        private ClientMethodDescriptor<ReqT, ResT> cmd;
```

Comment:
Because there were too many fields I followed this pattern. I had also added a comment as why I did it this way.

Anyways, I have changed it to follow the same pattern as other builders.